scripts/street_sign_recognizer.py

```diff
@@ -43,20 +132,34 @@ def sign_bounding_box(self):
         (left_top, right_bottom) where left_top and right_bottom are tuples of (x_pixel, y_pixel)
             defining topleft and bottomright corners of the bounding box
         """
-        # TODO: YOUR SOLUTION HERE
-        left_top = (200, 200)
-        right_bottom = (400, 400)
-        return left_top, right_bottom
+        # Find contours based on the binary filtered image
+        contours, hierarchy = cv2.findContours(self.filt_image, cv2.RETR_TREE, \
+           cv2.CHAIN_APPROX_SIMPLE)
+
+        # Define bounding rectangle around contour
+        if (contours):
+            x, y, w, h = cv2.boundingRect(contours[0])
+            left_top = (x, y)
+            right_bottom = (x+w, y+h)
+            return left_top, right_bottom
+        else:
+            # If no contours were found
+            return 0, 0
 
     def run(self):
         """ The main run loop"""
         r = rospy.Rate(10)
         while not rospy.is_shutdown():
-            if not self.cv_image is None:
-                print "here"
-                # creates a window and displays the image for X milliseconds
-                cv2.imshow('video_window', self.cv_image)
-                cv2.waitKey(5)
+            # Create windows to show all specified image streams
+            if (not self.bgr_image is None):
+                cv2.imshow('video_window', self.bgr_image)
+            if (not self.filt_image is None):
+                cv2.imshow('filt_window', self.filt_image)
+            if (not self.cropped_sign_grayscale is None):
+                cv2.imshow('cropped_grayscale_window', self.cropped_sign_grayscale)
+            if (not self.template_matcher.img_T is None):
+                cv2.imshow('img_T_window', self.template_matcher.img_T)
```

Comment:
Nice, I like how you made img_T an attribute to be accessible outside of the _compute_prediction function.  Now I feel bad for not thinking of this as a clean way to keep imshow's all in one place..